src/dmd/identifier.d

```diff
@@ -57,28 +64,28 @@ public:
 
     override bool equals(RootObject o) const
     {
-        return this == o || strncmp(name, o.toChars(), len + 1) == 0;
+        return this == o || name == o.toString();
     }
 
     override int compare(RootObject o) const
     {
-        return strncmp(name, o.toChars(), len + 1);
+        return strncmp(name.ptr, o.toChars(), name.length + 1);
     }
 
 nothrow:
     override void print() const
     {
-        fprintf(stderr, "%s", name);
+        fprintf(stderr, "%.*s", name.length, name.ptr);
```

Comment:
Must cast `name.length` to `int`.

Reply:
see documentation on `printf` format strings.

Reply:
Are those print method actually useful ? Shouldn't we just use `toChars` / `toString` everywhere ?

Reply:
Asking because if not, I might submit a separate PR to get rid of them

Reply:
Probably not. Mixing I/O into a class is usually a failure of encapsulation.